lib/Basic/Assertions.cpp

```diff
@@ -35,16 +35,7 @@ int CONDITIONAL_ASSERT_Global_enable_flag =
   0; // TODO: Default to `on` in debug builds
 #endif
 
-void ASSERT_failure(const char *expr, const char *file, int line, const char *func) {
-  // Only print the last component of `file`
-  const char *f = file;
```

Comment:
This is no longer needed because __FILE_NAME__ only includes the last component.